From b85ff55555fb3b564613b449d8630832c73e948e Mon Sep 17 00:00:00 2001
From: Philip Balister <philip@opensdr.com>
Date: Thu, 4 Aug 2016 10:41:16 -0400
Subject: [PATCH] Fix lib64 detection for cross builds.

OpenEmbedded sets CMAKE_INSTALL_LIBDIR so we can directly detect if
the target wants to install uhd in the lib64 directory.

Signed-off-by: Philip Balister <philip@opensdr.com>
---
 host/CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/host/CMakeLists.txt b/host/CMakeLists.txt
index 12cb916..6d31182 100644
--- a/host/CMakeLists.txt
+++ b/host/CMakeLists.txt
@@ -48,6 +48,14 @@ INCLUDE(UHDPackage)   #setup cpack
 IF(NOT DEFINED LIB_SUFFIX AND REDHAT AND CMAKE_SYSTEM_PROCESSOR MATCHES "64$")
     SET(LIB_SUFFIX 64)
 ENDIF()
+
+########################################################################
+# Detect /lib versus /lib64
+########################################################################
+if (CMAKE_INSTALL_LIBDIR MATCHES lib64)
+    set(LIB_SUFFIX 64)
+endif()
+
 SET(LIB_SUFFIX ${LIB_SUFFIX} CACHE STRING "lib directory suffix")
 SET(RUNTIME_DIR bin)
 SET(LIBRARY_DIR lib${LIB_SUFFIX})
-- 
2.5.5

